tests/tools/test_clang.lua

```diff
@@ -122,3 +122,16 @@ function suite.cxxflags_onSanitizeFuzzer()
 	prepare()
 	test.contains({ "-fsanitize=fuzzer" }, clang.getcxxflags(cfg))
 end
+
+--
+-- Check the handling of `getrelative` hook.
+--
+
+function suite.onGetRelative()
+	toolset "clang"
+	prepare()
+	local toolset = p.config.toolset(cfg)
+	print(toolset)
+	test.isnotnil(toolset.getrelative)
```

Comment:
Your test is really unit,
and I think it doesn't test what you want to check, i.e `clang.some_function_which_uses_getrelative()` calls actually `clang.getrelative` (especially with gcc redirection)
And I think that test would currently fail :-(

Reply:
I meant, if user (or premake.ninja) does `clang.getrelative = function (cfg, path) return p.workspace.getrelative(cfg.workspace, value) end`
then `clang.getincludedirs(cfg.includedirs, ..)` would call  `gcc.getincludedirs(cfg.includedirs, ..)` which will call
`gcc.getrelative` and not `clang.getrelative`.

As bug happens with premake.ninja, can you test your changes with problematic case (project location != workspace location) with clang toolset? (https://github.com/Jarod42/premake-sample-projects/tree/location/projects/project-location is one such project (not the main branch))

Reply:
A priori, I don't see issue with `p.tools.getrelative`